contracts/contracts/validator-registry/middleware/MevCommitMiddleware.sol

```diff
@@ -0,0 +1,607 @@
+    /// @return 0 if the blsPubkey is not in the set.
+    function _getPositionInValset(bytes calldata blsPubkey,
+        address vault, address operator) internal view returns (uint256) {
+        return _vaultAndOperatorToValset[vault][operator].position(blsPubkey);
+    }
+
+    function _isValidatorReadyToDeregister(bytes calldata blsPubkey) internal view returns (bool) {
+        return validatorRecords[blsPubkey].deregRequestOccurrence.exists && 
+            block.timestamp > slashPeriodSeconds + validatorRecords[blsPubkey].deregRequestOccurrence.timestamp;
+    }
+
+    function _isOperatorReadyToDeregister(address operator) internal view returns (bool) {
+        return operatorRecords[operator].deregRequestOccurrence.exists && 
+            block.timestamp > slashPeriodSeconds + operatorRecords[operator].deregRequestOccurrence.timestamp;
+    }
+
+    function _isVaultReadyToDeregister(address vault) internal view returns (bool) {
+        return vaultRecords[vault].deregRequestOccurrence.exists && 
+            block.timestamp > slashPeriodSeconds + vaultRecords[vault].deregRequestOccurrence.timestamp;
+    }
+
+    function _getSubnetwork() internal view returns (bytes32) {
+        return Subnetwork.subnetwork(network, SUBNETWORK_ID);
+    }
+
+    function _getSlashableVals(address vault, address operator) internal view returns (uint256) {
+        IBaseDelegator delegator = IBaseDelegator(IVault(vault).delegator());
+        uint256 allocatedStake = delegator.stake(_getSubnetwork(), operator);
+        uint256 slashAmount = vaultRecords[vault].slashAmount;
+        return allocatedStake / slashAmount;
+    }
+
+    // TODO: need to unit test
+    function _allValidatorsAreSlashable(address vault, address operator) internal view returns (bool) {
+        uint256 slashableVals = _getSlashableVals(vault, operator);
+        uint256 numVals = _vaultAndOperatorToValset[vault][operator].length();
+        return slashableVals >= numVals;
+    }
+
+    function _isValidatorSlashable(bytes calldata blsPubkey, address vault, address operator) internal view returns (bool) {
+        uint256 slashableVals = _getSlashableVals(vault, operator);
+        uint256 position = _getPositionInValset(blsPubkey, vault, operator);
+        require(position != 0, ValidatorNotInValset(blsPubkey, vault, operator));
+        return position <= slashableVals; // position is 1-indexed
+    }
+
+    function _potentialSlashableVals(address vault, address operator) internal view returns (uint256) {
+        uint256 slashableVals = _getSlashableVals(vault, operator);
+        uint256 alreadyRegistered = _vaultAndOperatorToValset[vault][operator].length();
+        if (slashableVals < alreadyRegistered) {
+            return 0;
+        }
+        return slashableVals - alreadyRegistered;
+    }
+    
+    function _isValidatorOptedIn(bytes calldata blsPubkey) internal view returns (bool) {
+        if (!validatorRecords[blsPubkey].exists) {
```

Comment:
It will be a little bit more effective to save `validatorRecords[blsPubkey]` to the local variable (`validatorRecord`)